File: apps/plant_shop_java/Structure_en_application_distribuée/plant_shop_java_javalin_distribuée/utils/src/PasswordUtil.java
package util;

import org.mindrot.jbcrypt.BCrypt;

/**
 * Utilitaire de hachage de mots de passe BCrypt.
 */
public final class PasswordUtil {

    /**
 * Constructeur privé - classe utilitaire.
 */
private PasswordUtil() {}

    /**
 * Hash un mot de passe avec BCrypt.
 * @param plainTextPassword Mot de passe en clair
 * @return Hash BCrypt
 */
public static String hashPassword(String plainTextPassword) {
        return BCrypt.hashpw(plainTextPassword, BCrypt.gensalt());
    }

    public static boolean checkPassword(String plainTextPassword, String hashedPassword) {
        if (plainTextPassword == null || hashedPassword == null) {
            return false;
        }
        try {
            return BCrypt.checkpw(plainTextPassword, hashedPassword);
        } catch (IllegalArgumentException e) {
            System.err.println("Avertissement : hash BCrypt invalide.");
            return false;
        }
    }
}

File: apps/plant_shop_c-sharp/plant_shop_asp_dapper/Utils/PasswordUtil.cs
namespace plant_shop_asp_dapper.Utils
{
    /// <summary>
    /// Utilitaire de hachage BCrypt.
    /// </summary>    // (Identique au Projet 2)
    public static class PasswordUtil
    {
        /// <summary>
        /// Hache un mot de passe avec BCrypt.
        /// </summary>
        /// <param name="password">Mot de passe en clair.</param>
        /// <returns>Hash BCrypt.</returns>
        public static string HashPassword(string password)
        {
            return BCrypt.Net.BCrypt.HashPassword(password);
        }

        public static bool CheckPassword(string password, string hashedPassword)
        {
            try
            {
                return BCrypt.Net.BCrypt.Verify(password, hashedPassword);
            }
            catch
            {
                return false;
            }
        }
    }
}
